### ISSUE TITLE
Feature request: allow passing custom attributes to initialize method (web)

### ISSUE THREAD

**Issue by Frank3K** [deepak786/intercom_flutter#466]
## Description

In the Intercom JavaScript SDK, it is possible to pass additional properties to the `boot` method, not just the `appId`, as is currently done in `intercom_flutter_web`. This could allow developers to initialize Intercom with more user-specific data.

For reference, here’s an example of how the `boot` method can accept additional properties:

```js
Intercom('boot', {  
    app_id: 'abc12345',  
    email: 'john.doe@example.com',
    created_at: 1234567890,
    name: 'John Doe',
    user_id: '9876'
});
```

Source: [Intercom SDK Documentation: `boot` Method](https://developers.intercom.com/installing-intercom/web/methods#intercomboot-intercomsettings)

Currently, the `intercom_flutter_web` package only passes the `appId` during boot, as shown here:

https://github.com/v3rm0n/intercom_flutter/blob/6e332b768ee81d0620fdead8cb18f799568d434d/intercom_flutter_web/lib/intercom_flutter_web.dart#L62-L66

### Feature Request

Would it be possible to add support for passing additional properties to the `boot` method? While I know it’s possible to inject these properties using `window.intercomSettings`, I’d prefer to handle this within Flutter for consistency.

**Comment by deepak786** [comment]
You can use [updateUser](https://pub.dev/documentation/intercom_flutter/latest/intercom_flutter/Intercom/updateUser.html) API to pass the additional data and custom attributes.

**Comment by Frank3K** [comment]
> You can use the [updateUser](https://pub.dev/documentation/intercom_flutter/latest/intercom_flutter/Intercom/updateUser.html) API to pass the additional data and custom attributes.

Yes, I’m aware of the `updateUser` option. However, in our use case, some data is already available during the application's bootstrap, so it would be beneficial to submit this information directly during the initialization of Intercom. This would avoid an additional `ping` request to Intercom, making the initialization more efficient.

I’ve tried calling `updateUser` immediately after awaiting the `initialize` method, but unfortunately, the `updateUser` call does not seem to send a `ping` request in that case. I'm still investigating the cause—it could be due to Intercom not being fully initialized at that point, or perhaps the call is being debounced.

Would you be open to a PR that adds this functionality, allowing properties to be passed during the initial `boot` as described in the feature request?

**Comment by deepak786** [comment]
If we allow passing some additional data only from the web during the initialization then it will be inconsistent with other platforms (Android and iOS). So I don't recommend this.

The ideal way is to chain the updateUser call after the initialize.

**Comment by Frank3K** [comment]
> If we allow passing some additional data only from the web during the initialization then it will be inconsistent with other platforms (Android and iOS). So I don't recommend this.
> 
> The ideal way is to chain the updateUser call after the initialize.

Thanks for your answer. I understand. I'll try to debug why the immediate `updateUser` does not fire.
